lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -52,46 +55,48 @@ module AutoEncrypter
       #
       # @raise [ ArgumentError ] If required options are missing or incorrectly
       #   formatted.
-      def setup_encrypter(options = {})
-        opts = set_default_options(options.dup)
-
-        unless opts[:key_vault_client]
-          # If no key vault client is passed in, create one by copying the
-          # Mongo::Client used for encryption. Update options so that key vault
-          # client does not perform auto-encryption/decryption, and keep a reference
-          # to it so it is destroyed later.
-          opts[:key_vault_client] = self.with({ auto_encryption_options: nil })
-        end
-
-        mongocryptd_client_monitoring_io = opts.delete(:mongocryptd_client_monitoring_io)
-        mongocryptd_client_monitoring_io = true if mongocryptd_client_monitoring_io.nil?
+      def initialize(options)
+        @options = set_default_options(options).freeze
 
-        super(opts)
+        @crypt_handle = Crypt::Handle.new(
+          @options[:kms_providers],
+          schema_map: @options[:schema_map]
+        )
 
-        @key_vault_client = opts[:key_vault_client]
+        @key_vault_client = @options[:key_vault_client]
 
         # Set server selection timeout to 1 to prevent the client waiting for a
         # long timeout before spawning mongocryptd
         @mongocryptd_client = Client.new(
-          @encryption_options[:mongocryptd_uri],
-          monitoring_io: mongocryptd_client_monitoring_io,
+          @options[:extra_options][:mongocryptd_uri],
+          monitoring_io: @options[:client].options[:monitoring_io],
           server_selection_timeout: 1,
         )
 
         @encryption_io = EncryptionIO.new(
-          client: self,
+          client: @options[:client],
           mongocryptd_client: @mongocryptd_client,
-          key_vault_collection: build_key_vault_collection
+          key_vault_namespace: @options[:key_vault_namespace],
+          key_vault_client: @key_vault_client,
+          mongocryptd_options: @options[:extra_options]
         )
       end
 
-      # Encrypt a database command
+      # Whether this encrypter should perform encryption (returns false if
+      # the :bypass_auto_encryption option is set to true).
+      #
+      # @return [ Boolean ] Whether to perform encryption.
+      def should_encrypt
```

Comment:
Since this is a predicate method it should have a `?` at the end, and perhaps `encrypt?` would be sufficient.

Reply:
Good point! Thanks.